RPIPs/RPIP-43.md

```diff
@@ -1,172 +0,0 @@
----
-rpip: 43
-title: Megapools
-description: A single contract per node operator that can be used as an Ethereum withdrawal address for multiple validators. Also allows for ETH-only node operation.
-author: Valdorff (@Valdorff)
-discussions-to: TBD
-status: Draft
-type: Protocol
-category: Core
-created: 2024-03-05
-requires: 30, 46, eip-4788
-tags: tokenomics-2024, tokenomics-content
----
-
-## Abstract
-This proposal drastically reduces the gas to add validators and distribute rewards for them. It does so by creating a single contract that can be used as the Ethereum withdrawal address for any number of validators. In addition, the contract will facilitate the application of node-level penalties, as opposed to the current state where penalties are assigned and processed at the minipool level. Note that this proposal also enables ETH-only Node operation.
-
-## Specification
-
-### Megapool Contracts
-
-This specification introduces a type of smart contract called a "megapool" that
-SHALL serve as the target of Beacon Chain withdrawal credentials for Rocket Pool
-validators deposited after this RPIP is implemented.
-
-- A Node Operator SHALL be able to deploy at most one megapool contract
-    - This MAY be combined with the deposit of their first validator
-- Megapool contracts MUST be upgradable using a proxy pattern; a megapool's
-  proxy implementation contract SHALL be called the megapool's "delegate"
-- Legacy minipool deposits (ie, deposits of validators with withdrawal
-  credentials that are not a megapool) SHALL be disabled
```

Comment:
Should we use stronger language here and link to the definition anytime it is used in other RPIPs? Maybe change:
> - Legacy minipool deposits (ie, deposits of validators with withdrawal credentials that are not a megapool) SHALL be disabled

to something like:
> - We use the term "legacy minipool" to refer to a Rocket Pool validator with withdrawal credentials that are not a megapool
> - Legacy minipool deposits SHALL be disabled

"Legacy minipool" is also used in RPIP42, RPIP46, and RPIP59:
![image](https://github.com/rocket-pool/RPIPs/assets/123328428/386bdb4a-d1d9-4563-ae32-6ffb154f1d0d)

Reply:
Probably makes sense to unify this. IMO define legacy minipool once in RPIP-43, most of the others have 43 as a dependency anyway, so can be assumed to be that definition.